### PR TITLE
adds basic testing

### PR DESCRIPTION
**Pull request by gregcaporaso**
this is needed to confirm that some lower level changes that I'm working on aren't breaking things. 

@ebolyen, could you review this one?

### CODE REVIEW

File: q2_diversity/__init__.py [hunk 1]
```diff
@@ -6,8 +6,8 @@
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
 
-__version__ = "0.0.0-dev"
-
 from ._beta import beta_diversity
 
+__version__ = "0.0.0-dev"
```
Comment (ebolyen) [inline q2_diversity/__init__.py]: pip doesn't handle `0.0.0` well nor dashes, change to `0.0.1dev`

File: q2_diversity/tests/test_beta.py [hunk 2]
```diff
@@ -5,3 +5,68 @@
...
+            for id2 in actual.ids:
+                npt.assert_almost_equal(actual[id1, id2], expected[id1, id2])
+
+    def test_non_phylogenetic_invalid_input(self):
+        t = Table(np.array([[0, 1, 3], [1, 1, 2]]),
+                  ['O1', 'O2'],
+                  ['S1', 'S2', 'S3'])
+        tree = skbio.TreeNode.read(io.StringIO(
+            '((O1:0.25, O2:0.50):0.25, O3:0.75)root;'))
+        self.assertRaises(TypeError, beta_diversity, 'braycurtis', t,
```
Comment (ebolyen) [inline q2_diversity/tests/test_beta.py]:
```python
with self.assertRaises(TypeError):
    beta_diversity('bray_curtis', t, phylogeny=tree)
```

File: q2_diversity/tests/test_beta.py [hunk 3]
```diff
@@ -5,3 +5,68 @@
...
+                          phylogeny=tree)
+
+    def test_phylogenetic_invalid_input(self):
+        t = Table(np.array([[0, 1, 3], [1, 1, 2]]),
+                  ['O1', 'O2'],
+                  ['S1', 'S2', 'S3'])
+        self.assertRaises(TypeError, beta_diversity, 'unweighted_unifrac', t)
```
Comment (ebolyen) [inline q2_diversity/tests/test_beta.py]: Use `with` like above.

File: q2_diversity/tests/test_beta.py [hunk 4]
```diff
@@ -5,3 +5,68 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
+
+from unittest import TestCase, main
```
Comment (ebolyen) [inline q2_diversity/tests/test_beta.py]: We tend to just import `unittest`, but it's not that important.

**Comment by ebolyen** [comment]
@gregcaporaso, done reviewing, only a few comments.

File: q2_diversity/__init__.py (hunk 1 above)
Comment (gregcaporaso) [inline q2_diversity/__init__.py]: This hasn't changed (just moved location in this file). Note this is what we're doing in the other plugins right now:
https://github.com/qiime2/q2-feature-table/blob/master/q2_feature_table/__init__.py#L15
https://github.com/qiime2/q2-types/blob/master/q2_types/__init__.py#L12

Mind if we just leave it that way for the moment, and change it in all the repos at the same time?

Comment (ebolyen) [inline q2_diversity/__init__.py]: sounds fine

**Comment by ebolyen** [comment]
Thanks @gregcaporaso!

**Pull request merged**